### PR TITLE
Add admin-api to unload a specific topic

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation

Many times we need to unload specific topic so, it can recover/reload with update config/state.

### Modifications

- Added `unload topic` admin api
- update the doc

### Result

Now, we can unload topic using admin api.

### CODE REVIEW

**Comment by rdhabalia** [comment]
> should we also add for non-persistent topics?

yes, will add it.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/NonPersistentTopics.java [hunk 1]
```diff
@@ -129,6 +129,18 @@ public void createPartitionedTopic(@PathParam("property") String property, @Path
             throw new RestException(e);
         }
     }
+
+    @PUT
+    @Path("/{property}/{cluster}/{namespace}/{destination}/unload")
+    @ApiOperation(value = "Unload a topic")
+    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission") })
```
Comment (jai1) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/NonPersistentTopics.java]: @ApiResponse(code = 404, message = "Topic does not exist"),

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PersistentTopics.java [hunk 2]
```diff
@@ -311,6 +311,33 @@
...
+    /**
+     * Unload a topic asynchronously.
+     * <p>
+     *
+     * @param destination
+     *            Destination name
+     *
+     * @return a future that can be used to track when the topic is unloaded
+     */
+    CompletableFuture<Void> unloadAsync(String destination);
```
Comment (jai1) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PersistentTopics.java]: Should we start providing just async versions and expect users to just call a get on the future to get blocking versions.

**Comment by rdhabalia** [comment]
added non-persistent unloading and documentation.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/NonPersistentTopics.java (hunk 1 above)
Comment (rdhabalia) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/NonPersistentTopics.java]: yes, fixed it.

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PersistentTopics.java (hunk 2 above)
Comment (rdhabalia) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PersistentTopics.java]: actually, this api is also used by cli-tool which requires blocking call. I think that's the reason why we have multiple api.

**Comment by rdhabalia** [comment]
retest this please

**Comment by rdhabalia** [comment]
retest this plesae

**Comment by rdhabalia** [comment]
retest this please

**Pull request merged**